FIX: Account for additional lines in cisco_nxos_bgp_summary

asik8888 opened the pull request:
Few lines in cisco_nxos_show_ip_bgp_summary output error out while parsing. The following changes can help with reading those additional lines seen in "show ip bgp summary" commands.  

Files modified:
cisco_nxos_show_ip_bgp_summary.textfsm
cisco_nxos_show_ip_bgp_summary.raw
cisco_nxos_show_ip_bgp_summary.yml

mjbear commented:
Hello @asik8888
Thank you for opening a PR.
The community here was welcoming to me so I intend to pay it forward here.

There are a few items missing from your PR that mean it needs some work before the maintainers will consider merging it. Please allow me to explain.

Please provide a PR description. It is important to provide a brief summary of what your changes accomplish.

There are several bits of documentation in [dev docs](https://github.com/networktocode/ntc-templates/tree/master/docs/dev) and [user docs](https://github.com/networktocode/ntc-templates/tree/master/docs/user)
Specifically:
* [Contributing guidelines](https://github.com/networktocode/ntc-templates/tree/master/docs/dev/contributing.md)
* [Test data is a requirement](https://github.com/networktocode/ntc-templates/tree/master/docs/user/faq.md#i-simply-want-to-add-my-template-to-the-project-i-do-not-want-to-add-all-of-these-tests-and-controls-can-i-just-do-so)
* Details on setting up your [development environment](https://github.com/networktocode/ntc-templates/blob/master/docs/dev/dev_environment.md)
    * [Helper scripts for parsing the raw output](https://github.com/networktocode/ntc-templates/tree/master/docs/dev/dev_parser.md#development-helper-scripts) you provide

I hope this helps guide you.
If anything needs clarified after you've looked at the documentation, please ask. :+1:

mjbear commented on ntc_templates/templates/cisco_nxos_show_ip_bgp_summary.textfsm:
```diff
@@ -17,10 +17,15 @@ Start
   ^${BGP_NEIGH}\s+\d+\s+${NEIGH_AS}\s+${MSG_RCVD}\s+${MSG_SENT}\s+\d+\s+${IN_QUEUE}\s+${OUT_QUEUE}\s+${UP_DOWN}\s+${STATE_PFXRCD}\s*$$ -> Record
   ^${BGP_NEIGH}\s+\d+\s+${NEIGH_AS}
   ^\s+${MSG_RCVD}\s+${MSG_SENT}\s+\d+\s+${IN_QUEUE}\s+${OUT_QUEUE}\s+${UP_DOWN}\s+${STATE_PFXRCD}\s*$$ -> Record
+  #ignoring all lines ending in bytes of memory
```
```suggestion
  # ignoring all lines ending in bytes of memory
```

mjbear reviewed (commented):
Regarding the first suggestion, it's more readable to put a space between the # symbol and the comment.

On the second suggestion, it's best to use regexes to account for white space just in case it was to ever change down the road.

(Flexibility by way of regexes means we hopefully shouldn't find it break often as vendors possibly change their output formatting. It's also a middle ground :balance_scale: between flexibility and making the regexes overly complex.)

asik8888 commented:
Hello Mjbear, 

Thank you for the helpful suggestions, I appreciate your input in helping me. I have resolved the conflicts from the above two suggestions, please let me know if I can do anything further for this PR

mjbear commented:
> Few lines in "show ip bgp summary" output error out while parsing due to the following lines not being handled correctly in the template. These changes can help with reading those additional lines seen in "show ip bgp summary" commands.

@asik8888 
You might update the title and description a bit to call out which template.

It's tough to really tell without the test data so I'll probably botch my "title" suggestion below until I see the test data. :sweat_smile: 

_Title:_
Account for additional lines in nxos bgp summary

_Body:_
Few lines in **cisco_nxos** "show ip bgp summary" output error out while parsing due to the following lines not being handled correctly in the template. These changes can help with reading those additional lines seen in "show ip bgp summary" commands.

asik8888 commented:
Hello @mjbear,

Added the raw data and yml file to the tests folder. Invoke test, passed.
![test](https://github.com/user-attachments/assets/da46ad9f-ce55-4cd0-8c94-70fcaffd08bb)

Updated the title and body. Let me know if you would like it in a different format or any other changes. 

Thank you for the help!

mjbear commented:
> Hello @mjbear,
> 
> Added the raw data and yml file to the tests folder. Invoke test, passed. ![test](https://private-user-images.githubusercontent.com/37921102/377745322-da46ad9f-ce55-4cd0-8c94-70fcaffd08bb.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyNTg1NDgsIm5iZiI6MTcyOTI1ODI0OCwicGF0aCI6Ii8zNzkyMTEwMi8zNzc3NDUzMjItZGE0NmFkOWYtY2U1NS00Y2QwLThjOTQtNzBmY2FmZmQwOGJiLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE4VDEzMzA0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTljYTZkMmQwNDlmNDk1OWFjOTE2NmI4ODhhOTI4ZTc1MzA4ZDVlODE2NDk2NWQwNTdkMmUwNDcyYzFiMTNjNzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JN6apnygInDn2uE96-AJVzw6EnvnbBVhkPD-ewCMRiA)
> 
> Updated the title and body. Let me know if you would like it in a different format or any other changes.
> 
> Thank you for the help!

Hello @asik8888,

Apologies for not providing more guidance about the test data.
Adding new test files is oftentimes better than modifying existing ones. (more on that below) :wink:

I would would strongly recommend retaining the existing tests/cisco_nxos/show_ip_bgp_summary/cisco_nxos_show_ip_bgp_summary.raw in its original form.

Then from there add another file with your new test data (example: tests/cisco_nxos/show_ip_bgp_summary/cisco_nxos_show_ip_bgp_summary2.raw).

:bulb: **This way the test data includes not only the old example, but also your new example. This helps provide a broader test base so the project testing has a chance at catching any unexpected errors/regressions.**

Hope that helps!

mjbear commented:
@asik8888 
I hope you're doing well.

I dropped a PR over on your fork so you should be able to easily merge in those tweaks and update this PR.
Please review https://github.com/asik8888/ntc-templates/pull/1 when you have a chance.
Thank you!

asik8888 commented:
Thank you for input and creating a pull request. sorry for the delay in adding the required files in order. Got busy on something else. I apologize. Have merged the pull request.

asik8888 commented:
Hi @jmcgill298 

was wondering when this will be released?

mjbear commented:
> Hi @jmcgill298
> 
> was wondering when this will be released?

Hello @asik8888 
I'll provide some observations here, but I can't speak for the NTC team as to when the next ntc-templates release will be.

The changes you brought forward were merged into the master branch so it is available, but just not yet part of a versioned release. You could `git clone` the repo and point your scripts at that clone if you chose to.

Now as a longer term item a release is certainly more desirable. Seeing as how [the last release (7.3.0) was two weeks ago](https://github.com/networktocode/ntc-templates/releases/tag/v7.3.0) it is possible (though no guarantee) that a release may be coming in a week or two.

asik8888 commented:
Thank you for the update @Michael bear. Appreciate it



On Mon, Nov 4, 2024 at 6:27 AM Michael Bear ***@***.***>
wrote:

> Hi @jmcgill298 <https://github.com/jmcgill298>
>
> was wondering when this will be released?
>
> Hello @asik8888 <https://github.com/asik8888>
> I'll provide some observations here, but I can't speak for the NTC team as
> to when the next ntc-templates release will be.
>
> The changes you brought forward were merged into the master branch so it
> is available, but just not yet part of a versioned release. You could git
> clone the repo and point your scripts at that clone if you chose to.
>
> Now as a longer term item a release is certainly more desirable. Seeing as
> how the last release (7.3.0) was two weeks ago
> <https://github.com/networktocode/ntc-templates/releases/tag/v7.3.0> it
> is possible (though no guarantee) that a release may be coming in a week or
> two.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/networktocode/ntc-templates/pull/1877#issuecomment-2454586302>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJBKCTV7XHQTG524VDERZA3Z65KYRAVCNFSM6AAAAABP5PJ7FSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINJUGU4DMMZQGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

jmcgill298 commented:
its out now

asik8888 commented:
Thank you Jacob!


On Mon, Nov 4, 2024 at 9:43 AM Jacob McGill ***@***.***>
wrote:

> its out now
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/networktocode/ntc-templates/pull/1877#issuecomment-2455052504>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJBKCTQDTTRMBASTFF3SYDLZ66BZHAVCNFSM6AAAAABP5PJ7FSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINJVGA2TENJQGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
